### PR TITLE
chore: update liveness UX

### PR DESCRIPTION
**Pull request by thaddmt**
#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/__tests__/LivenessCameraModule.test.tsx [hunk 1]
```diff
@@ -15,16 +15,23 @@
...
   LivenessCameraModule,
+  selectFaceMatchPercentage,
+  selectFaceMatchState,
+  selectSelectableDevices,
+  selectSelectedDeviceId,
   selectVideoConstraints,
   selectVideoStream,
 } from '../LivenessCameraModule';
-import { FaceMatchState } from '../../service';
+import { FaceMatchState, drawStaticOval } from '../../service';
```
[the remark this answers is not in the document]
Comment (reesscot) [reply packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/__tests__/LivenessCameraModule.test.tsx:25]: Please address these unused imporst

File: examples/next/pages/ui/components/liveness/components/LivenessInlineResults.tsx [hunk 2]
```diff
@@ -28,12 +28,27 @@ export default function LivenessInlineResults({
...
         </Badge>
       </Text>
 
+      {!isLive && (
+        <Text as="span">
+          <Text fontWeight="bold">Tips to pass the video check:</Text>
+          <ol>
+            <li>Maximize you screen&apos;s brightness</li>
+            <li>
+              Avoud very bright lightinh conditions, such as direct sunkight.
```
Comment (reesscot) [inline examples/next/pages/ui/components/liveness/components/LivenessInlineResults.tsx:43]:
```suggestion
              Avoid very bright lighting conditions, such as direct sunlight.
```

File: examples/next/pages/ui/components/liveness/components/LivenessInlineResults.tsx [hunk 3]
```diff
@@ -28,12 +28,27 @@ export default function LivenessInlineResults({
...
+            </li>
+            <li>
+              Remove sunglasses, mask, hat, or anything blocking you face.
```
Comment (reesscot) [inline examples/next/pages/ui/components/liveness/components/LivenessInlineResults.tsx:46]:
```suggestion
              Remove sunglasses, mask, hat, or anything blocking your face.
```

File: packages/react/src/hooks/useTheme.ts [hunk 4]
```diff
@@ -27,3 +28,11 @@ export const useTheme = (): WebTheme => {
   const context = React.useContext(ThemeContext);
   return getThemeFromContext(context);
 };
+
+/**
+ * Internal use only
+ */
+export const useColorMode = (): ColorMode | undefined => {
```
Comment (thaddmt) [inline packages/react/src/hooks/useTheme.ts:35]: Added this hook to see when the color mode has changed, made it internal only for now since it is just used in liveness to see when the color mode has changed

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/DefaultStartScreenComponents.tsx [hunk 5]
```diff
@@ -46,58 +30,45 @@ export const DefaultPhotosensitiveWarning = ({
...
 
-export const DefaultInstructions = ({
-  headingText,
-  goodFitCaptionText,
-  goodFitAltText,
-  tooFarCaptionText,
-  tooFarAltText,
-  steps,
-}: DefaultInstructionsProps): JSX.Element => {
+export const DefaultRecordingIcon = ({
```
Comment (thaddmt) [inline packages/react-liveness/src/components/FaceLivenessDetector/shared/DefaultStartScreenComponents.tsx:50]: We removed the original start screen with instructions and added the recording icon and cancel button as default components that can be overridden

File: packages/react-liveness/src/components/FaceLivenessDetector/service/utils/liveness.ts [hunk 6]
```diff
@@ -182,6 +182,42 @@ export function getStaticLivenessOvalDetails({
   };
 }
 
+export function drawStaticOval(
```
Comment (thaddmt) [inline packages/react-liveness/src/components/FaceLivenessDetector/service/utils/liveness.ts:185]: Helper method for drawing a static oval. Used to redraw the static oval when the color mode changes since we need to re-compute the style for the background color

File: packages/react-liveness/src/components/FaceLivenessDetector/service/machine/index.ts [hunk 7]
```diff
@@ -154,11 +155,22 @@ export const livenessMachine = createMachine<LivenessContext, LivenessEvent>(
...
             cond: 'hasDOMAndCameraDetails',
           },
-          // setting this to check every 500 ms sometimes caused detectFaceBeforeStart to be called twice
-          500: { target: 'waitForDOMAndCameraDetails' },
+          10: { target: 'waitForDOMAndCameraDetails' },
+        },
+      },
+      start: {
+        entry: ['drawStaticOval', 'initializeFaceDetector'],
+        always: [
```
Comment (thaddmt) [inline packages/react-liveness/src/components/FaceLivenessDetector/service/machine/index.ts:166]: Once we reach the start state we wait for the user to click the `begin check` button. If the dev added the `skipStartScreen` flag then we can skip this

File: packages/react/src/index.ts [hunk 8]
```diff
@@ -1,5 +1,5 @@
 export * from './components';
-export { useBreakpointValue, useTheme } from './hooks';
+export { useBreakpointValue, useTheme, useColorMode } from './hooks';
```
Comment (reesscot) [inline packages/react/src/index.ts:2]: Is this API currently public? If not, can we export via /internals?

Comment (reesscot) [reply packages/react/src/index.ts:2]: export it from here: https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/src/internal.ts

File: packages/react-liveness/src/components/FaceLivenessDetector/displayText.ts [hunk 9]
```diff
@@ -72,12 +73,12 @@ export const defaultLivenessDisplayText: Required<LivenessDisplayText> = {
   instructionsHeaderHeadingText: 'Liveness check',
   instructionsHeaderBodyText:
     'You will go through a face verification process to prove that you are a real person.',
-  instructionsBeginCheckText: 'Begin check',
+  instructionsBeginCheckText: 'Start video check',
```
Comment (reesscot) [inline packages/react-liveness/src/components/FaceLivenessDetector/displayText.ts:76]: So nice that changes like this are no longer breaking

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Toast.tsx [hunk 10]
```diff
@@ -1,28 +1,42 @@
...
+  isInitial = false,
   ...rest
 }) => {
+  const { tokens } = useTheme();
   return (
     <View
       className={`${LivenessClassNames.Toast} ${LivenessClassNames.Toast}--${variation} ${LivenessClassNames.Toast}--${size}`}
-      maxWidth={{ base: '100%', small: '70%' }}
+      maxWidth={{ base: '100%' }}
+      {...((variation === 'primary' || variation === 'error') && {
```
Comment (reesscot) [inline packages/react-liveness/src/components/FaceLivenessDetector/shared/Toast.tsx:25]: nit: could probably declare this styling object outside the function so a new object isn't generated each time.

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx [hunk 11]
```diff
@@ -87,18 +95,28 @@ export const Hint: React.FC<HintProps> = ({ hintDisplayText }) => {
...
       }
 
       // Specifically checking for false here because initially the value is undefined and we do not want to show the instruction
       if (
         isCheckFaceDistanceBeforeRecording &&
         isFaceFarEnoughBeforeRecordingState === false
       ) {
-        return <Toast>{hintDisplayText.hintTooCloseText}</Toast>;
+        return (
+          <Toast size="large" variation="primary">
```
Comment (reesscot) [inline packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx:116]: nonblocking nit: Might be nice to declare each of these Toast components as named Components outside the function since they don't  seem to change. This one could be called `TooCloseHint` or something.

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/DefaultStartScreenComponents.tsx [hunk 12]
```diff
@@ -46,58 +30,45 @@ export const DefaultPhotosensitiveWarning = ({
   return (
     <Flex
       className={`${ComponentClassName.Alert} ${LivenessClassNames.StartScreenWarning}`}
+      style={{ zIndex: '3' }}
```
Comment (reesscot) [inline packages/react-liveness/src/components/FaceLivenessDetector/shared/DefaultStartScreenComponents.tsx:33]: This styling could go on StartScreenWarning class?

File: packages/react-liveness/src/components/FaceLivenessDetector/service/utils/__tests__/streamProvider.test.ts [hunk 13]
```diff
@@ -20,20 +19,26 @@ Object.defineProperty(window, 'TextDecoder', {
   value: TextDecoder,
 });
 
-const mockGet = jest.fn().mockImplementation(() => {
+jest.mock('aws-amplify/auth', () => {
+  const originalModule = jest.requireActual('aws-amplify/auth');
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-return
```
Comment (reesscot) [inline packages/react-liveness/src/components/FaceLivenessDetector/service/utils/__tests__/streamProvider.test.ts:24]: Is this still needed?

**Review by reesscot — DISMISSED** [review]
LGTM, just a few nits. Also need to add migration docs for the props that changed before release, but it's ok if that's a separate PR.

File: packages/react-liveness/src/components/FaceLivenessDetector/__tests__/FaceLivenessDetector.test.tsx [hunk 14]
```diff
@@ -94,19 +94,8 @@ describe('FaceLivenessDetector', () => {
 
   it('should show the check screen if disableInstructionScreen is true', () => {
     render(
-      <FaceLivenessDetector {...defaultProps} disableInstructionScreen={true} />
+      <FaceLivenessDetector {...defaultProps} disableStartScreen={true} />
     );
     expect(screen.queryByTestId(livenessCheckTestId)).toBeInTheDocument();
   });
-
-  it('should not show the instruction if disableInstructionScreen is true and xstate is at the start', async () => {
```
[the remark this answers is not in the document]
Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/__tests__/FaceLivenessDetector.test.tsx:102]: yep, we dont show the start screen here anymore

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx (hunk 11 above)
Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/shared/Hint.tsx:116]: Will do in a follow up, i'll add a note to myself

File: packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx [hunk 15]
```diff
@@ -151,109 +227,205 @@ export const LivenessCameraModule = (
...
-        </Overlay>
-      </View>
-    </Flex>
+            {isRecording &&
+            !isFlashingFreshness &&
+            showMatchIndicatorStates.includes(faceMatchState!) ? (
+              <MemoizedMatchIndicator
+                percentage={Math.ceil(faceMatchPercentage!)}
+              />
+            ) : null}
```
Comment (calebpollman) [inline packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx:379]: Concerned the conditional logic going on here is fragile and well be difficult to maintain.

Only calling out here but imo applies to the rest of the file, there is a lot of opportunity to cross cut concerns by scoping components to specific needs

File: packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx [hunk 16]
```diff
@@ -151,109 +227,205 @@ export const LivenessCameraModule = (
...
+                  >
+                    Camera:
+                  </Label>
+                  <SelectField
+                    id="amplify-liveness-camera-select"
+                    label="Camera"
+                    labelHidden
+                    value={selectedDeviceId}
+                    // eslint-disable-next-line @typescript-eslint/no-misused-promises
+                    onChange={onCameraChange}
```
Comment (calebpollman) [inline packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx:404]: Can `onCameraChange` be treated as fire and forget?

File: packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCheck.tsx [hunk 17]
```diff
@@ -12,31 +12,34 @@ import {
...
 );
 
 interface LivenessCheckProps {
+  instructionDisplayText: Required<InstructionDisplayText>;
   hintDisplayText: Required<HintDisplayText>;
   cameraDisplayText: Required<CameraDisplayText>;
   streamDisplayText: Required<StreamDisplayText>;
   errorDisplayText: Required<ErrorDisplayText>;
-  components?: CheckScreenComponents;
+  components?: FaceLivenessDetectorComponents;
```
Comment (calebpollman) [inline packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCheck.tsx:38]: Prefer scoped `components` interfaces to enforce a separation of concerns between slots

File: packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCheck.tsx [hunk 18]
```diff
@@ -143,7 +146,7 @@ export const LivenessCheck: React.FC<LivenessCheckProps> = ({
               ? cameraMinSpecificationsHeadingText
               : cameraNotFoundHeadingText}
           </Text>
-          <Text maxWidth={300}>
+          <Text maxWidth={420}>
```
Comment (calebpollman) [inline packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCheck.tsx:149]: Possible to remove a magic number by assigning `420` to a meaningfully named constant?

File: packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/__tests__/LivenessCameraModule.test.tsx [hunk 19]
```diff
@@ -69,9 +83,11 @@ describe('LivenessCameraModule', () => {
     isCheckingCamera = false;
     isNotRecording = false;
     isRecording = false;
+    isStart = false;
 
     jest.clearAllMocks();
     jest.clearAllTimers();
+    mockUseLivenessSelector.mockClear();
```
Comment (calebpollman) [inline packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/__tests__/LivenessCameraModule.test.tsx:90]: Doesn't `jest.clearAllMocks` clear all mocks 😅

File: .changeset/pretty-tables-live.md [hunk 20]
```diff
@@ -0,0 +1,5 @@
+---
+"@aws-amplify/ui-react-liveness": major
+---
+
+chore: update liveness UX
```
Comment (dbanksdesign) [inline .changeset/pretty-tables-live.md:5]:
```suggestion
breaking(liveness): update liveness UX
```

And maybe add a bit more description about what the UX changes are

File: packages/react-liveness/src/components/FaceLivenessDetector/service/machine/index.ts (hunk 7 above)
Comment (calebpollman) [reply packages/react-liveness/src/components/FaceLivenessDetector/service/machine/index.ts:166]: Excluding the start screen is a UI concern and extraneously increases complexity in the state machine imo

File: packages/react-liveness/src/components/FaceLivenessDetector/service/machine/index.ts [hunk 21]
```diff
@@ -447,6 +458,22 @@ export const livenessMachine = createMachine<LivenessContext, LivenessEvent>(
           freshnessColorEl: event.data?.freshnessColorEl,
         }),
       }),
+      updateDeviceAndStream: assign({
+        videoAssociatedParams: (context, event) => {
+          localStorage.setItem(CAMERA_ID_KEY, event.data?.newDeviceId);
```
Comment (calebpollman) [inline packages/react-liveness/src/components/FaceLivenessDetector/service/machine/index.ts:463]: This feels like a side effect?

File: packages/ui/src/theme/css/component/liveness.scss [hunk 22]
```diff
@@ -6,7 +6,8 @@
 }
 
 .amplify-liveness-cancel-button {
-  background-color: var(--amplify-colors-background-primary);
+  background-color: var(--amplify-colors-white);
+  color: hsl(190, 95%, 30%);
```
Comment (dbanksdesign) [inline packages/ui/src/theme/css/component/liveness.scss:10]: We should use a CSS variable here

File: packages/ui/src/theme/css/component/liveness.scss [hunk 23]
```diff
@@ -97,12 +98,16 @@
 .amplify-liveness-recording-icon {
   flex-direction: column;
   align-items: center;
-  background-color: var(--amplify-colors-background-primary);
+  background-color: var(--amplify-colors-white);
```
Comment (dbanksdesign) [inline packages/ui/src/theme/css/component/liveness.scss:101]: Is this change intentional? background-primary references white so there isn't really an effect by default, but would make the theme API harder to grok because the color white has no semantic meaning

File: packages/react-liveness/src/components/FaceLivenessDetector/service/utils/liveness.ts [hunk 24]
```diff
@@ -203,10 +241,15 @@ export function drawLivenessOvalInCanvas({
...
   if (ctx) {
+    ctx.restore();
     ctx.clearRect(0, 0, canvasWidth, canvasHeight);
 
     // fill the canvas with a transparent rectangle
-    ctx.fillStyle = 'rgba(255, 255, 255, 1.0)';
+    ctx.fillStyle = getComputedStyle(canvas).getPropertyValue(
+      isStartScreen
+        ? '--amplify-colors-background-primary'
+        : '--amplify-colors-white'
```
Comment (calebpollman) [inline packages/react-liveness/src/components/FaceLivenessDetector/service/utils/liveness.ts:251]: Think this reads like it should be evaluated in a ui component. Does it need to be here?

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Toast.tsx [hunk 25]
```diff
@@ -1,28 +1,46 @@
...
   children,
+  isInitial = false,
   ...rest
 }) => {
+  const { tokens } = useTheme();
   return (
     <View
       className={`${LivenessClassNames.Toast} ${LivenessClassNames.Toast}--${variation} ${LivenessClassNames.Toast}--${size}`}
-      maxWidth={{ base: '100%', small: '70%' }}
+      maxWidth={maxWidthStyle}
```
Comment (dbanksdesign) [inline packages/react-liveness/src/components/FaceLivenessDetector/shared/Toast.tsx:32]: `maxWidthStyle` set above is `{ base: '100%' }`, which is equivalent of saying `maxWidth="100%"` but doesn't have any extra style computations based on breakpoint.

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Toast.tsx [hunk 26]
```diff
@@ -1,28 +1,46 @@
...
+  isInitial?: boolean;
 }
 
+const toastStyle = {
+  borderRadius: 'unset',
+  fontSize: 'xxl',
+  padding: '0 var(--amplify-space-xs)',
+};
+
+const maxWidthStyle = { base: '100%' };
```
Comment (dbanksdesign) [inline packages/react-liveness/src/components/FaceLivenessDetector/shared/Toast.tsx:19]: Can we move all this stuff to CSS and classnames? It is kind of hard to understand what is happening

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/DefaultStartScreenComponents.tsx [hunk 27]
```diff
@@ -1,39 +1,23 @@
...
-      </View>
-    </View>
-  );
-};
-
 interface DefaultPhotosensitiveWarningProps {
-  headingText: string;
+  headingText?: string;
   bodyText: string;
   infoText: string;
```
Comment (calebpollman) [inline packages/react-liveness/src/components/FaceLivenessDetector/shared/DefaultStartScreenComponents.tsx:22]: Why is headingText now optional?

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/DefaultStartScreenComponents.tsx [hunk 28]
```diff
@@ -46,58 +30,45 @@ export const DefaultPhotosensitiveWarning = ({
   return (
     <Flex
       className={`${ComponentClassName.Alert} ${LivenessClassNames.StartScreenWarning}`}
+      style={{ zIndex: '3' }}
     >
       <View flex="1">
-        <View className={ComponentClassName.AlertHeading}>{headingText}</View>
+        {headingText && (
```
Comment (dbanksdesign) [inline packages/react-liveness/src/components/FaceLivenessDetector/shared/DefaultStartScreenComponents.tsx:36]: Nit: prefer ternary for conditional rendering

File: packages/react-liveness/src/components/FaceLivenessDetector/service/utils/liveness.ts (hunk 24 above)
Comment (dbanksdesign) [inline packages/react-liveness/src/components/FaceLivenessDetector/service/utils/liveness.ts:251]: These are the same by default, what is this logic doing?

File: packages/react-liveness/src/components/FaceLivenessDetector/service/types/liveness.ts [hunk 29]
```diff
@@ -34,7 +34,7 @@ export interface FaceLivenessDetectorCoreProps {
   /**
    * Optional parameter for the disabling the Start/Get Ready Screen, default: false
    */
-  disableInstructionScreen?: boolean;
+  disableStartScreen?: boolean;
```
Comment (dbanksdesign) [inline packages/react-liveness/src/components/FaceLivenessDetector/service/types/liveness.ts:37]: Is this a breaking change? Can this be called out in the changeset and added to a migration guide?

**Review by dbanksdesign — COMMENTED** [review]
Left some comments and questions about styling and docs

**Review by calebpollman — COMMENTED** [review]
Have concerns with the how large the primary UI components are becoming and adding additional state machine properties

File: packages/ui/src/theme/css/component/liveness.scss (hunk 23 above)
Comment (thaddmt) [reply packages/ui/src/theme/css/component/liveness.scss:101]: This was intentional to keep it white because Rekognition asked that the color not change with the color theme. They wanted it always to stay white

File: packages/react-liveness/src/components/FaceLivenessDetector/service/utils/liveness.ts (hunk 24 above)
Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/service/utils/liveness.ts:251]: The way rekognition wanted it to be was that on the start screen we would respect the theme background color.

When we are in the check we would always use white to ensure that we have a higher chance of passing the freshness color check. This logic makes it so that when changing to dark mode we ensure that it is always white on the check screen.

File: packages/react-liveness/src/components/FaceLivenessDetector/service/types/liveness.ts (hunk 29 above)
Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/service/types/liveness.ts:37]: Already added to migration guide, will call out in changeset

File: packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx [hunk 30]
```diff
@@ -151,109 +227,205 @@ export const LivenessCameraModule = (
...
-                errorState,
-                overrideErrorDisplayText: errorDisplayText,
-              })}
-            </ErrorView>
+          <video
+            ref={videoRef}
+            muted
+            autoPlay
+            playsInline
+            style={{ transform: 'scaleX(-1)' }}
```
[the remark this answers is not in the document]
Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx:318]: nope, fixed

File: packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx [hunk 31]
```diff
@@ -151,109 +227,205 @@ export const LivenessCameraModule = (
...
+            <CancelButton cancelLivenessCheckText={cancelLivenessCheckText} />
           )}
 
-          {/* 
+          <Overlay
+            anchorOrigin={{
+              horizontal: 'center',
+              vertical:
+                isRecording && !isFlashingFreshness ? 'start' : 'space-between',
+            }}
```
[the remark this answers is not in the document]
Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx:348]: yep, will add

File: packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx (hunk 16 above)
Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx:404]: Just tested and yes it can, good call

File: packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/__tests__/LivenessCameraModule.test.tsx (hunk 19 above)
Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/__tests__/LivenessCameraModule.test.tsx:90]: yep think i added this in frustration while testing lol

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/DefaultStartScreenComponents.tsx (hunk 27 above)
Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/shared/DefaultStartScreenComponents.tsx:22]: honestly not sure why i chagned this, good call, changing back

File: packages/react-liveness/src/components/FaceLivenessDetector/service/utils/liveness.ts (hunk 24 above)
Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/service/utils/liveness.ts:251]: Not necessarily, just added it here because I didn't want to do this style computation at the caller of drawLivenessOvalInCanvas which happens in a few places now

File: packages/react-liveness/src/components/FaceLivenessDetector/service/machine/index.ts (hunk 21 above)
Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/service/machine/index.ts:463]: Can be removed, was just added as a way to persist camera preference

File: packages/react-liveness/src/components/FaceLivenessDetector/service/utils/__tests__/streamProvider.test.ts (hunk 13 above)
Comment (calebpollman) [reply packages/react-liveness/src/components/FaceLivenessDetector/service/utils/__tests__/streamProvider.test.ts]: Will need to pass a generic, or could replace with spying

File: packages/react-liveness/src/components/FaceLivenessDetector/service/machine/index.ts (hunk 21 above)
Comment (ayush987goyal) [reply packages/react-liveness/src/components/FaceLivenessDetector/service/machine/index.ts:463]: I dont think it can be removed? We do need to persist the camera preference.

File: docs/src/pages/[platform]/getting-started/migration/migration.react.mdx [hunk 32]
```diff
@@ -681,6 +681,55 @@ The latest version of the `Authenticator` has a completely different set of CSS
...
+      )}
+    </ThemeProvider>
+  );
+);
+
+```
+
+#### CDN CSP Policy
+
+The 3.x version of the `FaceLivenessDetector` has been updated to use the latest version of TensorFlow and Blazefacee thus the default CDN's have changed to a new URL. If your application has an existing CSP policy, ensure your policy allows https://cdn.liveness.rekognition.amazonaws.com. If you are using a custom CDN make sure to update your CDN versions to match [@tensorflow/tfjs-backend-wasm](https://www.npmjs.com/package/@tensorflow/tfjs-backend-wasm/v/4.11.0) and [@tensorflow-models/face-detection](https://www.npmjs.com/package/@tensorflow-models/face-detection/v/1.0.2).Please look over the [Liveness Config](../connected-components/liveness/customization#facelivenessdetectorconfig) section for more information.
```
Comment (reesscot) [inline docs/src/pages/[platform]/getting-started/migration/migration.react.mdx:731]:
```suggestion
The 3.x version of the `FaceLivenessDetector` has been updated to use the latest version of TensorFlow and Blazeface, thus the default CDN paths have changed. If your application has an existing CSP policy, ensure your policy allows https://cdn.liveness.rekognition.amazonaws.com. If you are using a custom CDN make sure to update your CDN versions to match [@tensorflow/tfjs-backend-wasm](https://www.npmjs.com/package/@tensorflow/tfjs-backend-wasm/v/4.11.0) and [@tensorflow-models/face-detection](https://www.npmjs.com/package/@tensorflow-models/face-detection/v/1.0.2).Please look over the [Liveness Config](../connected-components/liveness/customization#facelivenessdetectorconfig) section for more information.
```

**Comment by reesscot** [comment]
re: https://github.com/aws-amplify/amplify-ui/pull/4728#discussion_r1395139611
I think this deserves further discussion, can we address moving the start screen UI logic to UI in a follow up PR if it's not a breaking change?

File: packages/react-liveness/src/components/FaceLivenessDetector/service/utils/liveness.ts (hunk 24 above)
Comment (reesscot) [reply packages/react-liveness/src/components/FaceLivenessDetector/service/utils/liveness.ts:251]: Should we call this out in documentation, so customers aren't confused by this behavior?

Comment (reesscot) [reply packages/react-liveness/src/components/FaceLivenessDetector/service/utils/liveness.ts:251]: Could we put it in a function we use at the call site instead?

Comment (dbanksdesign) [reply packages/react-liveness/src/components/FaceLivenessDetector/service/utils/liveness.ts:251]: If we want it to always be white should we just hard-code it to be `#fff` ? Because technically, customers can override the color-white token to be any color.

File: packages/react-liveness/src/components/FaceLivenessDetector/service/machine/index.ts (hunk 21 above)
Comment (calebpollman) [reply packages/react-liveness/src/components/FaceLivenessDetector/service/machine/index.ts:463]: Persisting the camera preference is fine, but usage of side effects in the state machine removes determinism from the machine itself

File: packages/react-liveness/src/components/FaceLivenessDetector/service/utils/liveness.ts (hunk 24 above)
Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/service/utils/liveness.ts:251]: I can update that. Will also update the color for the cancel button to be hard coded as well since they always want that to stay the same color

**Review by dbanksdesign — DISMISSED** [review]

**Review by reesscot — APPROVED** [review]

File: packages/react/src/components/ThemeProvider/ThemeContext.tsx [hunk 33]
```diff
@@ -1,11 +1,14 @@
...
+import { ColorMode } from './ThemeProvider';
 
 export interface ThemeContextType {
   theme: WebTheme;
+  colorMode?: ColorMode;
 }
 
 export const ThemeContext = React.createContext<ThemeContextType>({
   theme: createTheme(),
+  colorMode: undefined,
```
Comment (calebpollman) [inline packages/react/src/components/ThemeProvider/ThemeContext.tsx:13]: This is a far reaching API change, can we remove it

File: packages/react-liveness/src/components/FaceLivenessDetector/types/classNames.ts [hunk 34]
```diff
@@ -34,6 +34,9 @@ export enum LivenessClassNames {
   PopoverAnchorSecondary = 'amplify-liveness-popover__anchor-secondary',
   RecordingIconContainer = 'amplify-liveness-recording-icon-container',
   RecordingIcon = 'amplify-liveness-recording-icon',
+  StartScreenCameraSelect = 'amplify-liveness-start-screen-camera-select',
+  StartScreenCameraSelectContainer = 'amplify-liveness-start-screen-camera-select__container',
+  StartScreenCameraWaiting = 'amplify-liveness-start-screen-camera-waiting',
```
Comment (calebpollman) [inline packages/react-liveness/src/components/FaceLivenessDetector/types/classNames.ts:39]: Are these public?

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Toast.tsx [hunk 35]
```diff
@@ -1,28 +1,36 @@
...
       className={`${LivenessClassNames.Toast} ${LivenessClassNames.Toast}--${variation} ${LivenessClassNames.Toast}--${size}`}
-      maxWidth={{ base: '100%', small: '70%' }}
+      {...(isInitial && { backgroundColor: tokens.colors.background.primary })}
       {...rest}
     >
       <Flex className={LivenessClassNames.ToastContainer}>
-        <Flex className={LivenessClassNames.ToastMessage}>{children}</Flex>
+        <Flex
+          className={LivenessClassNames.ToastMessage}
+          {...(isInitial ? { color: tokens.colors.font.primary } : {})}
```
Comment (calebpollman) [inline packages/react-liveness/src/components/FaceLivenessDetector/shared/Toast.tsx:30]: Think this can just be pass `color={isInitial ? tokens.colors.font.primary : undefined}`

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Toast.tsx [hunk 36]
```diff
@@ -1,28 +1,36 @@
...
-import { Flex, View, ViewProps } from '@aws-amplify/ui-react';
+import { Flex, View, ViewProps, useTheme } from '@aws-amplify/ui-react';
 
 import { LivenessClassNames } from '../types/classNames';
 
 interface ToastProps extends ViewProps {
   variation?: 'default' | 'primary' | 'error';
   size?: 'medium' | 'large';
   heading?: string;
+  isInitial?: boolean;
```
Comment (calebpollman) [inline packages/react-liveness/src/components/FaceLivenessDetector/shared/Toast.tsx:10]: What does `isInitial` mean?

File: packages/react-liveness/src/components/FaceLivenessDetector/types/classNames.ts (hunk 34 above)
Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/types/classNames.ts:39]: We don't document this anywhere

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Overlay.tsx [hunk 37]
```diff
@@ -2,26 +2,23 @@ import * as React from 'react';
...
 
-interface AnchorOrigin {
-  horizontal: 'start' | 'center' | 'end';
-  vertical: 'start' | 'center' | 'end' | 'space-between';
-}
-
 interface OverlayProps extends FlexProps {
-  anchorOrigin?: AnchorOrigin;
+  horizontal?: 'start' | 'center' | 'end';
+  vertical?: 'start' | 'center' | 'end' | 'space-between';
```
Comment (calebpollman) [inline packages/react-liveness/src/components/FaceLivenessDetector/shared/Overlay.tsx:7]: Are these properties not already available in `FlexProps`?

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Toast.tsx (hunk 36 above)
Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/shared/Toast.tsx:10]: Indicates it's the toast used in the start screen which has a different styling

**Review by dbanksdesign — APPROVED** [review]

File: packages/react-liveness/src/components/FaceLivenessDetector/shared/Toast.tsx (hunk 36 above)
Comment (calebpollman) [reply packages/react-liveness/src/components/FaceLivenessDetector/shared/Toast.tsx:10]: Can we just pass the style down?

**Review by calebpollman — COMMENTED** [review]
:ship:
